noaa_metrics/aggregate_logs.py

```diff
@@ -16,59 +20,278 @@
 )
 
 
+def get_available_memory_gb():
+    """Get available system memory in GB."""
+    return psutil.virtual_memory().available / (1024**3)
+
+
+def get_file_size_mb(filepath: Path) -> float:
+    """Get file size in MB."""
+    try:
+        return os.path.getsize(filepath) / (1024 * 1024)
```

Comment:
In get_available_memory_gb(), the denominator is 1024^3.  
For "similarity", I'd suggest either making this denominator 1024^2 or the other 1024*1024*1024.